design/one-pager-default-resource-class.md

```diff
@@ -88,66 +116,237 @@ spec:
+
+In this model, a claim will now look as follows:
 
 ```yaml
-apiVersion: storage.crossplane.io/v1alpha1
-kind: NoSQLInstance
+apiVersion: database.crossplane.io/v1alpha1
+kind: MySQLInstance
 metadata:
-  name: dynamo-claim
-  namespace: demo
+  name: mysql-claim
 spec:
-  ...
+  policyRef:
+    name: standard-mysql
+    namespace: crossplane-system
+  writeConnectionSecretToRef:
+    name: rdsmysql
+  engineVersion: "5.6"
 ```
 
-Or in the more specific case:
+With an arbitrary number of potential `MySQLInstancePolicy` objects that could
+be referenced:
 
 ```yaml
-apiVersion: storage.crossplane.io/v1alpha1
-kind: DynamoDBInstance
+apiVersion: database.crossplane.io/v1alpha1
+kind: MySQLInstancePolicy
 metadata:
-  name: dynamo-claim
-  namespace: demo
-spec:
-  ...
+  name: standard-mysql
+  namespace: crossplane-system
+classRef:
+  kind: RDSInstanceClass
+  apiVersion: database.aws.crossplane.io/v1alpha1
+  name: standard-mysql
+  namespace: crossplane-system
 ```
 
+### Denote Default via Label
+
+This feature is similar to the original default class model in that it uses
+labels to specify which class of service to use as default when a `policyRef` is
+omitted. For each `namespace`, there must be only one policy that has the
+`default` label:
+
+```yaml
+apiVersion: database.crossplane.io/v1alpha1
+kind: MySQLInstancePolicy
+metadata:
+  name: standard-mysql
+  namespace: crossplane-system
+  labels:
+    default: true
+classRef:
+  kind: RDSInstanceClass
+  apiVersion: database.aws.crossplane.io/v1alpha1
+  name: standard-mysql
+  namespace: crossplane-system
+```
+
+To implement this functionality, the following steps must be taken:
+
+1. Update [default class controller
+   predicates](https://github.com/crossplaneio/crossplane-runtime/blob/e4d61ee2805af680baf16fc2a1d8f79538d0f9bb/pkg/resource/predicates.go#L60)
+   to accept resource claims that do not have a `policyRef`.
+1. Update the [shared default class
+   reconciler](https://github.com/crossplaneio/crossplane-runtime/blob/e4d61ee2805af680baf16fc2a1d8f79538d0f9bb/pkg/resource/defaultclass.go#L106)
+   to set the `policyRef` of a claim to the `Policy` in its `namespace` with the
+   `default` label. If multiple `Policies` with the `default` label exist in the
+   `namespace`, the controller should fail to reconcile.
+
+Both of the above changes should be implemented in `crossplane-runtime` and will
+likely not involve any changes to the actual implementations of the default
+class controllers in core Crossplane.
+
+## Future Considerations
+
+Introducing this new layer to the Crossplane dynamic provisioning pattern allows
+for possible future expansion of the functionality of `Policies`, which may
+include "intelligent" `Policy` defaulting for claims, or referencing multiple
+resource classes in the same `Policy` and picking one based on specific
+parameters defined in the claim.
+
+It is also likely that strongly-typed claims could be introduced at the provider
```

Comment:
> strongly-typed claims could be introduced

How are we imagining this working? Is this talking about using something like bringing back the `classRef` field in a claim, or is it talking about something like a new claim type which only works with a single provider?

Reply:
The latter. I think it is definitely possible that we will want non-portable resources to still be able to be dynamically provisioned.